07-memory/user_space/main.c

```diff
@@ -0,0 +1,109 @@
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+#define NVALUE 64
+#define NSTAT 10
+
+static void malloc_inv(int ind, int cal, int p1)
+{
+	clock_t tm, tf;
+	void *ptr;
+	double btm = 0;
+	double btf = 0;
+	double min_tm = 0;
+	double max_tm = 0;
+	double min_tf = 0;
+	double max_tf = 0;
+
+	double size = pow(2, ind);
+	if (p1) {
+		size++;
+	}
+
+	for (int i = 0; i <  NSTAT; i++) {
+
+		if (cal == 2) {
+			tm = clock();
+			ptr = alloca(size * sizeof(void));
+			tm = clock() - tm;
+		} else if (cal == 1) {
+			tm = clock();
+			ptr = calloc(size, sizeof(void));
+			tm = clock() - tm;
+		} else {
+			tm = clock();
+			ptr = malloc(size * sizeof(void));
+			tm = clock() - tm;
+		}
+
+		if (ptr == NULL) {
+			printf("%2d %22.0f: Memory not allocated.\n", ind, size);
+			return;
+		}
+
+		btm += ((double)tm);
+		if ((double)tm < min_tm)
+			min_tm = (double)tm;
+		if ((double)tm > max_tm)
+			max_tm = (double)tm;
+
+		if (cal != 2) {
+			tf = clock();
+			free(ptr);
+			tf = clock() - tf;
+			btf += ((double)tf);
+
+			if ((double)tf < min_tf)
+				min_tf = (double)tf;
+			if ((double)tf > max_tf)
+				max_tf = (double)tf;
+		}
+
+	}
+	printf("%2d %22.0f %10.0f %10.0f %10.0f %10.0f %10.0f %10.0f\n",
```

Comment:
What is the unit for results? ms or ns or something else. Consider using CLOCKS_PER_SEC definition.